includes/utils/class-amp-dom-utils.php

```diff
@@ -231,14 +244,15 @@ public static function convert_amp_bind_attributes( $html ) {
 			return '<' . $tag_matches['name'] . $new_attrs . '>';
 		};
 
-		$html = preg_replace_callback(
+		$converted = preg_replace_callback(
 			// Match all start tags that probably contain a binding attribute.
-			'#<(?P<name>[a-zA-Z0-9_\-]+)(?P<attrs>\s+[^>]+\]=[^>]+)\s*>#',
+			'#<(?P<name>[a-zA-Z0-9_\-]+)(?P<attrs>\s[^>]+\]=[^>]+)>#',
 			$replace_callback,
 			$html
 		);
 
-		return $html;
+		// If the conversion failed, pass the original HTML as DOMDocument may still be able to use it.
```

Comment:
Maybe good to explain why it failed. Linking back the GitHub issue and mentioning backtracking would be good.